lib/iris/cube.py

```diff
@@ -2939,40 +2901,15 @@ def _order(array):
             if order and ma.isMaskedArray(data):
                 data_xml_element.setAttribute('mask_order',
                                               _order(data.mask))
-        else:
-            dtype = self.lazy_data().dtype
-        data_xml_element.setAttribute('dtype', dtype.name)
+            else:
+                dtype = self.lazy_data().dtype
+            data_xml_element.setAttribute('dtype', dtype.name)
```

Comment:
@lbdreyer I think you need to back this tab change out. 

Referring back to v1.13 the `else` on `line 2904` aligns with the `if` on `line2879`, and **not** the `if` on `line 2901`.

The previous logic makes sense, in that the `dtype` xml attribute is always set, and for non-lazy data extra xml metadata is set for the concrete data payload.